submissions/ses3332002/friends-app/app.js

```diff
@@ -6,51 +6,55 @@ const ageFromOption = document.querySelector("#age_from");
 const ageToOption = document.querySelector("#age_to");
 const searchNameOption = document.querySelector(".filter_name");
 const resetButton = document.querySelector(".reset_button");
+const renewButton = document.querySelector(".renew_button");
 const requestString = "https://randomuser.me/api/?nat=gb,us,fr&results=60&inc=gender,name,location,email,picture,dob";
 let receivedData = [];
-let dataToRender = [];
 let filterCriterionSex;
 let filterCriterionAge
 let filterCriterionName;
 let sortCriterion;
 hamburgerMenu.addEventListener("click", () => hamburgerMenu.classList.toggle("hamburger_menu-opened"));
-resetButton.addEventListener("click", resetApp);
+resetButton.addEventListener("click", prepareForm);
+renewButton.addEventListener("click", prepareList);
 optionForm.addEventListener("input", changeData);
-resetApp();
+prepareList();
   
-function resetApp() {
+function prepareForm() {
   filterCriterionSex = () => true;
   filterCriterionAge = () => true;
   filterCriterionName = () => true;
   sortCriterion = () => false;
   optionForm.reset();
-  
+  render(receivedData.slice(), sortCriterion, filterCriterionSex, filterCriterionAge, filterCriterionName);
+};
+
+function prepareList() {
   fetch(requestString)
   .then((response) => {
     return response.json();
   })
   .then((data) => {
     receivedData = data.results;
-    dataToRender = receivedData.slice();
-    render();
+    prepareForm();
   })
   .catch((err) => {
     mainArea.innerHTML = "There was some error, try again!";
     console.error("there was some error:", err);
   });
 };
 
-function render() {
+function sexString(age) {
+  const lastNumberOfAge = age.toString().slice(-1);
+  const numberForYearWord = "1";
+  const numbersForYearsWord = ["2", "3", "4"];
+  return (lastNumberOfAge == numberForYearWord) ? "год" : (numbersForYearsWord.includes(lastNumberOfAge)) ? "года" : "лет";
+};
+
+function render(arrayToShow, sortingFunction, filterFunction1, filterFunction2, filterFunction3) {
```

Comment:
The render function should only accept a list of users. Will it be filtered or sorted for it does not matter.
And you don't need to call filtering and sorting on the first render. I already told you about this.

Reply:
I just wanted to make my functions multipurpose. That is why there is no matter for render function which time it is called - the first time or not. But in the first time the sorting and filtering functions are "clear" (they return true or false, not real parameters).
What bad will happen in this way - if function runs the same the first time and the next, but the first time - with the initial filters?
Ok, I will remake it.

Reply:
I remade it